### PR TITLE
feat(auth): migrate from allauth-2fa to allauth[mfa]

### PR DESCRIPTION
**Pull request by marcpfuller**
### Issue

No issue logged yet.

### Description of the Change
This PR migrates our authentication system from the deprecated allauth-2fa package to the newer built-in MFA functionality in django-allauth. The changes include:

- Removing dependencies on allauth-2fa and its related components
- Implementing django-allauth's native MFA functionality for TOTP authentication
- Updating the authentication flow to use the new MFA interfaces
- Migrating existing user MFA device data to the new schema
- Updating templates and views to support the new authentication workflow
- The migration ensures that existing users with configured MFA devices will continue to have their MFA settings intact after the update, providing a seamless transition for end users.

### Alternate Designs
Several approaches were considered for this migration:

- Custom MFA implementation: Rejected in favor of using the officially supported solution
- Direct user data manipulation: Considered directly manipulating user data but chose a migration path that properly handles the transition
- Phased rollout: Evaluated running both systems in parallel but opted for a clean migration to avoid complexity
The implemented solution follows the recommended migration path from the django-allauth documentation, ensuring compatibility with future updates.

### Possible Drawbacks
- Minor changes to the MFA setup and authentication flow UI that users will need to adapt to
- Potential one-time re-authentication required for some users after the migration
- Slightly different session handling for MFA that might affect some edge-case workflows

### Verification Process
The changes were thoroughly tested to ensure proper functionality:

Unit Tests:

1. Build images: `docker compose -f local.yml build`
2. Run tests: `./ghostwriter-cli-macos test`
3. Tear down all containers and volumes: `docker compose -f local.yml down -v`

Manual tests:

1. Build images: `docker compose -f local.yml build`
2. Start Ghostwriter(local): `./ghostwriter-cli-macos install --dev`
3. Login into Admin and setup MFA device(DUO) 
4. Verify device is created
5. Log out of admin and  login
6. Verify device authentication view and able to login
7.  Create new user and enable `require mfa`
8. Login to new user account 
9. Verify device authentication
10. Delete device
11. Verify Device is deleted and verify that system requests device setup again (since `require mfa` is enabled)
12. Login to admin and delete Device
13. verify that device is deleted
14. Tear down containers and volumes: `docker compose -f local.yml down -v`
15. Enable reveal tokens update .env: `DJANGO_MFA_ALWAYS_REVEAL_BACKUP_TOKENS='true'`
16. Login to admin and add new MFA device
17. Verify that recovery tokens are displayed
18. Select `Generate Backup Tokens`
19. Verify new tokens have been generated

Release Notes
Migrated from allauth-2fa to django-allauth's to enable support for passkeys

### CODE REVIEW

File: ghostwriter/users/management/commands/migrate_totp_device.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                errors += 1
+                logger.error(f"Authenticator not found for user {totp.user_id}: {str(e)}")
+            except Authenticator.MultipleObjectsReturned as e:
+                # For unexpected database state
+                errors += 1
+                logger.error(f"Multiple authenticators found for user {totp.user_id}: {str(e)}")
+            except (IOError, OSError) as e:
+                # For file system errors (if any)
+                errors += 1
+                logger.error(f"I/O error for user {totp.user_id}: {str(e)}")
```
Comment (ColonelThirtyTwo) [inline ghostwriter/users/management/commands/migrate_totp_device.py]: Use `logger.exception` so that we get the traceback for these, which may be useful

**Review by ColonelThirtyTwo — COMMENTED** [review]
Looks good to me, don't see anything objectionable. Nice work.

File: ghostwriter/users/management/commands/migrate_totp_device.py (hunk 1 above)
Comment (marcpfuller) [reply ghostwriter/users/management/commands/migrate_totp_device.py]: yep,  I have made the change

**Comment by marcpfuller** [comment]
@ColonelThirtyTwo  I realized that I need to add in documentation for running the migrate command. I will get this added shortly

**Comment by marcpfuller** [comment]
> @ColonelThirtyTwo I realized that I need to add in documentation for running the migrate command. I will get this added shortly

Documentation has been added. Once [GW_CLI PR 20](https://github.com/GhostManager/Ghostwriter_CLI/pull/20) has merged Ill add the binaries to this PR.

**Review by ColonelThirtyTwo — APPROVED** [review]
